tests/transforms/test_color.py

```diff
@@ -33,12 +34,14 @@ def batch() -> dict[str, Tensor]:
     ],
 )
 def test_random_grayscale_sample(weights: Tensor, sample: dict[str, Tensor]) -> None:
-    aug = AugmentationSequential(RandomGrayscale(weights, p=1), data_keys=['image'])
+    aug = K.AugmentationSequential(
+        RandomGrayscale(weights, p=1), keepdim=True, data_keys=None
+    )
+    aug.keepdim = True
```

Comment:
Why is this line needed? Isn't this already set on the previous line?

Reply:
Gotcha. Can you add:
```suggestion
    # https://github.com/kornia/kornia/issues/2848
    aug.keepdim = True
```
Then we'll know why and eventually be able to remove this line.